Source/Core/Core/Scripting/Python/Pyr.h

```diff
@@ -0,0 +1,197 @@
+    return "l";
+  else if constexpr (std::is_same<T, signed long long>::value)
+    return "L";
+  else if constexpr (std::is_same<T, unsigned char>::value)
+    return "B";
+  else if constexpr (std::is_same<T, unsigned short>::value)
+    return "H";
+  else if constexpr (std::is_same<T, unsigned int>::value)
+    return "I";
+  else if constexpr (std::is_same<T, unsigned long>::value)
+    return "k";
+  else if constexpr (std::is_same<T, unsigned long long>::value)
+    return "K";
+  else if constexpr (std::is_same<T, float>::value)
+    return "f";
+  else if constexpr (std::is_same<T, double>::value)
+    return "d";
+  else if constexpr (std::is_same<T, PyObject*>::value)
+    return "O";
+  else if constexpr (std::is_same<T, char*>::value)
+    return "s";
+  else
+    static_assert(false, "unrecognized type, cannot convert to python format character");
+}
+}
+
+namespace PyrArg
+{
+template <typename T1>
+bool ParseTupleSimple(PyObject* args, T1* arg1)
+{
+  return PyArg_ParseTuple(args, Pyr::get_pyformat<T1>(), arg1);
+}
+
+template <typename T1, typename T2>
+bool ParseTupleSimple(PyObject* args, T1* arg1, T2* arg2)
+{
+  const std::string format =
+      std::string(Pyr::get_pyformat<T1>()) + std::string(Pyr::get_pyformat<T2>());
+  return PyArg_ParseTuple(args, format.c_str(), arg1, arg2);
+}
+}
+
+namespace PyrImport
+{
+inline Pyr::Object ImportModule(const char* module)
+{
+  return Pyr::wrap(PyImport_ImportModule(module));
+}
+}
+
+namespace PyrObject
+{
+inline Pyr::Object GetAttrString(const Pyr::Object& py_object, const char* attr)
+{
+  return Pyr::wrap(PyObject_GetAttrString(py_object.unwrap(), attr));
+}
+inline int SetAttrString(const Pyr::Object& py_object, const char* attr, const Pyr::Object& value)
+{
+  return PyObject_SetAttrString(py_object.unwrap(), attr, value.unwrap());
+}
+inline Pyr::Object CallFunction(const Pyr::Object& callable_object, const char* format)
+{
+  return Pyr::wrap(PyObject_CallFunction(callable_object.unwrap(), format));
+}
+inline Pyr::Object GetIter(const Pyr::Object& iterable_object)
+{
+  return Pyr::wrap(PyObject_GetIter(iterable_object.unwrap()));
+}
+}
+
+namespace PyrIter
+{
+inline Pyr::Object Next(const Pyr::Object& iterable_object)
+{
+  return Pyr::wrap(PyIter_Next(iterable_object.unwrap()));
+}
+}
+
+namespace PyrErr
+{
+inline const char* GetException()
+{
+  PyObject *ptype, *pvalue, *ptraceback;
+  PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+  PyObject* repr = PyObject_Repr(pvalue);
+  PyObject* pyStr = PyUnicode_AsEncodedString(repr, "utf-8", "Error ~");
+  return PyBytes_AS_STRING(pyStr);
+}
+}
+
+namespace PyrRun
+{
+inline Pyr::Object File(std::string file, PyObject* globals, PyObject* locals)
+{
+  return Pyr::wrap(PyRun_FileExFlags(fopen(file.c_str(), "r"), file.c_str(), Py_file_input, globals,
+                                     locals, true, nullptr));
+}
+}
+
+namespace PyrGILState
+{
+class GIL
+{
+  PyGILState_STATE gstate;
```

Comment:
Public interface before private interface. Class variables should also be prefixed with an `m_`